File: 0144-binary-tree-preorder-traversal/0144-binary-tree-preorder-traversal.java
class Solution {
    List<Integer> result;

    public List<Integer> preorderTraversal(TreeNode root) {
        result = new ArrayList<>();
        if (root == null) {
            return result;
        }
        traverse(root);
        return result;
    }

    private void traverse(TreeNode node) {
        result.add(node.val);
        if (node.left != null) {
            traverse(node.left);
        }
        if (node.right != null) {
            traverse(node.right);
        }
    }
}

File: 0144-binary-tree-preorder-traversal/0144-binary-tree-preorder-traversal.py
class Solution:
    def preorderTraversal(self, root: Optional[TreeNode]) -> List[int]:
        def traverse(node):
            result.append(node.val)
            if node.left:
                traverse(node.left)
            if node.right:
                traverse(node.right)


        result = list()

        if not root: return result
        traverse(root)
        return result